### PR TITLE
Added the ability to switch from ClusterRoleBinding to RoleBinding.

### PR DESCRIPTION
**Pull request by vfarcic**
**What this PR does / why we need it**:

Jenkins Chart allows specifying ClusterRoleBinding. That does not allow fine-tuning SA to give Jenkins permissions to access only specific Namespaces. This change allows users to switch from ClusterRoleBinding (default) to RoleBinding.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by jlegrone** [comment]
@vfarcic thanks, this will definitely help eliminate the need for creating rolebindings out of band or in a downstream chart.  To fix the tests, please increment the version in `Chart.yaml`, and add a new entry in the readme for `roleBindingKind`.

**Comment by vfarcic** [comment]
Done. Please let me know if there's anything else missing.

File: stable/jenkins/templates/rbac.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {{ if .Values.rbac.install }}
 {{- $serviceName := include "jenkins.fullname" . -}}
 apiVersion: rbac.authorization.k8s.io/{{ required "A valid .Values.rbac.apiVersion entry required!" .Values.rbac.apiVersion }}
-kind: ClusterRoleBinding
+kind: {{ .Values.rbac.roleBindingKind }}
```
Comment (lachie83) [inline stable/jenkins/templates/rbac.yaml:4]: If this is changed to rolebinding does the namespace key in the subjects array still work? Have you tested this?

Comment (vfarcic) [reply stable/jenkins/templates/rbac.yaml:4]: It does work, as far as I saw/tested. Are there some kind of automated tests I could extend?

Comment (vfarcic) [reply stable/jenkins/templates/rbac.yaml:4]: I did a few additional rounds of testing and it works without any issues.

**Comment by vfarcic** [comment]
It's done @dtzar . Is there anything else missing before merging this PR?

**Comment by dtzar** [comment]
@vfarcic [Per CircleCI](https://circleci.com/gh/kubernetes/charts/14855?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link), looks like you just need to bump the chart version.

File: stable/jenkins/templates/rbac.yaml (hunk 1 above)
Comment (carlossg) [reply stable/jenkins/templates/rbac.yaml:4]: `roleRef.kind` should be `Role` instead of `ClusterRole`

File: stable/jenkins/README.md [hunk 2]
```diff
@@ -78,6 +78,7 @@ The following tables list the configurable parameters of the Jenkins chart and t
 | `rbac.install`                    | Create service account and ClusterRoleBinding for Kubernetes plugin | `false`                                       |
 | `rbac.apiVersion`                 | RBAC API version                     | `v1beta1`                                                                    |
 | `rbac.roleRef`                    | Cluster role name to bind to         | `cluster-admin`                                                              |
```
Comment (carlossg) [inline stable/jenkins/README.md:81]: I'd change it to `Role name to bind to`

File: stable/jenkins/templates/rbac.yaml (hunk 1 above)
Comment (vfarcic) [reply stable/jenkins/templates/rbac.yaml:4]: I guess that there should be an option to use either of the two kinds. I can work on it as well but I'd keep that as a separate PR.

File: stable/jenkins/README.md (hunk 2 above)
Comment (vfarcic) [reply stable/jenkins/README.md:81]: I agree, but only after we add the option to customize Role kind.

**Comment by vfarcic** [comment]
If possible, I'd appreciate if we could merge this PR and keep the additional improvements separate. That way I could stop patching this particular issue.

**Comment by vfarcic** [comment]
@dtzar @lachie83 @jlegrone can we merge this PR?

**Comment by dtzar** [comment]
@vfarcic - I'd merge it if I had the permissions to ;)

**Comment by vfarcic** [comment]
I bumped the version as instructed by the test failure and I'm not sure what to do next. Can someone help me out from the process perspective?

**Comment by jlegrone** [comment]
@vfarcic we need an `/ok-to-test` or `/lgtm` from a project maintainer.  @lachie83 mind taking another look?

**Comment by vfarcic** [comment]
@jlegrone I'm a bit desperate to see this PR merged. It's a very simple change and I tested it quite a few times. Please let me know if there's anything I can do to speed things up.

**Comment by muehlburger** [comment]
I'd love to see this PR merged too.

**Comment by vfarcic** [comment]
@lachie83 ^^^

**Pull request merged**